WeakAurasOptions/OptionsFrames/TextEditor.lua

```diff
@@ -193,106 +304,346 @@ local function ConstructTextEditor(frame)
+    end
+  )
+
+  -- Make Snippets button (top right, near the line number)
+  local snippetsButton = CreateFrame("Button", "WASnippetsButton", group.frame, "UIPanelButtonTemplate")
+  snippetsButton:SetPoint("BOTTOMRIGHT", editor.frame, "TOPRIGHT", 0, -15)
+  snippetsButton:SetFrameLevel(group.frame:GetFrameLevel() + 2)
+  snippetsButton:SetHeight(20)
+  snippetsButton:SetWidth(100)
+  snippetsButton:SetText(L["Snippets"])
+  snippetsButton:RegisterForClicks("LeftButtonUp")
+
+  -- Get the saved snippets from SavedVars
+  WeakAurasOptionsSaved.savedSnippets = WeakAurasOptionsSaved.savedSnippets or {}
+  local savedSnippets = WeakAurasOptionsSaved.savedSnippets
+
+  -- function to build snippet selection list
+  local function UpdateSnippets(frame)
+    -- release first before rebuilding
+    frame:ReleaseChildren()
+    table.sort(
+      savedSnippets,
+      function(a, b)
+        return a.name < b.name
+      end
+    )
+
+    local heading1 = AceGUI:Create("Heading")
+    heading1:SetText(L["Premade Snippets"])
+    heading1:SetRelativeWidth(1)
+    frame:AddChild(heading1)
+
+    -- Iterate premade snippets and make buttons for them
+    for order, snippet in ipairs(premadeSnippets) do
+      local button = AceGUI:Create("WeakAurasSnippetButton")
+      button:SetTitle(snippet.name)
+      button:SetDescription(snippet.snippet)
+      button:SetCallback(
+        "OnClick",
+        function()
+          editor.editBox:Insert(snippet.snippet)
+          editor:SetFocus()
+        end
+      )
+      button:SetRelativeWidth(1)
+      frame:AddChild(button)
+    end
 
-  -- CTRL + S saves and closes, ESC cancels and closes
-  editor.editBox:HookScript("OnKeyDown", function(_, key)
-    if IsControlKeyDown() and key == "S" then
-      group:Close()
+    local heading2 = AceGUI:Create("Heading")
+    heading2:SetText(L["Your Saved Snippets"])
+    heading2:SetRelativeWidth(1)
+    frame:AddChild(heading2)
+
+    -- iterate saved snippets and make buttons
+    for order, snippet in ipairs(savedSnippets) do
+      local button = AceGUI:Create("WeakAurasSnippetButton")
+      button:SetTitle(snippet.name)
+      button:SetDescription(snippet.snippet)
+      button:SetEditable(true)
+      button:SetRelativeWidth(1)
+      button:SetCallback(
+        "OnClick",
+        function()
+          WeakAuras.editor.editBox:Insert(snippet.snippet)
+          WeakAuras.editor:SetFocus()
+        end
+      )
+      button.deleteButton:SetScript(
+        "OnClick",
+        function()
+          table.remove(savedSnippets, order)
+          UpdateSnippets(frame)
+        end
+      )
+      button:SetCallback(
+        "OnEnterPressed",
+        function()
+          local newName = button.renameEditBox:GetText()
+          if newName and #newName > 0 then
+            local found = false
+            for _, snippet in ipairs(savedSnippets) do
+              if snippet.name == newName then
+                found = true
```

Comment:
It would be nice if this behaved in a similar way as the aura list in the main window.